bigquery/tests/unit/test_client.py

```diff
@@ -3484,6 +3492,68 @@ def test_load_table_from_file_bad_mode(self):
         with pytest.raises(ValueError):
             client.load_table_from_file(file_obj, self.TABLE_REF)
 
+    @unittest.skipIf(pandas is None, 'Requires `pandas`')
+    @unittest.skipIf(pyarrow is None, 'Requires `pyarrow`')
+    def test_load_table_from_dataframe(self):
+        from google.cloud.bigquery.client import _DEFAULT_NUM_RETRIES
+        from google.cloud.bigquery import job
+
+        client = self._make_client()
+        records = [
+            {'name': 'Monty', 'age': 100},
+            {'name': 'Python', 'age': 60},
+        ]
+        dataframe = pandas.DataFrame(records)
```

Comment:
Let's use the `index` argument to DataFrame. Hopefully no extra columns beyond the data columns + index column are created. https://pandas.pydata.org/pandas-docs/stable/generated/pandas.DataFrame.html

Reply:
Added in https://github.com/GoogleCloudPlatform/google-cloud-python/pull/5387/commits/fe6d81dd431a35824d1a7e6fc7bc7328d2ffb44a